lightning/src/ln/channelmonitor.rs

```diff
@@ -1253,7 +1253,9 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
 			feerate_per_kw,
 			htlc_outputs,
 		});
-		self.onchain_tx_handler.provide_latest_local_tx(commitment_tx);
+		if let Err(_) = self.onchain_tx_handler.provide_latest_local_tx(commitment_tx) {
+			return Err(MonitorUpdateError("Local commitment signed has already been signed, no further update of LOCAL commitment transaction is allowed"));
```

Comment:
Shouldnt this occur before we update the local state in the ChannelMonitor itself?

Reply:
I moved it, but added a big note than you may use in the loosely future a watchtower with a different implementation monitoring backend, which doesn't reject update in this case and may broadcast the latest local signed commitment received as update (assuming it has key access).  And you want this watchtower to be able to claim outputs. That's really an edge-case so just a comment for now.